pulp_docker/tests/functional/api/test_recursive_remove.py

```diff
@@ -0,0 +1,319 @@
+# # coding=utf-8
+"""Tests that recursively remove docker content from repositories."""
+import unittest
+
+from pulp_smash import api, config
+from pulp_smash.pulp3.constants import REPO_PATH
+from pulp_smash.pulp3.utils import gen_repo, sync
+from requests.exceptions import HTTPError
+
+from pulp_docker.tests.functional.constants import (
+    DOCKER_TAG_PATH,
+    DOCKER_REMOTE_PATH,
+    DOCKER_RECURSIVE_ADD_PATH,
+    DOCKER_RECURSIVE_REMOVE_PATH,
+    DOCKERHUB_PULP_FIXTURE_1,
+)
+from pulp_docker.tests.functional.utils import gen_docker_remote
+
+
+class TestRecursiveRemove(unittest.TestCase):
+    """Test recursively removing docker content from a repository."""
```

Comment:
It is very helpful to add what a mention to the issue/feature that this test is related in case of failures later on.

Something like:

```.rst
    This test targets the following issue:

    `Pulp #4956 <https://pulp.plan.io/issues/4956>`_
```